exp/addons/controllers/clusterresourceset_controller_test.go

```diff
@@ -289,14 +285,18 @@ metadata:
 			return false
 		}, timeout).Should(BeTrue())
 
-		testConfigmap := &corev1.ConfigMap{
+		newConfigmap := &corev1.ConfigMap{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      newCMName,
 				Namespace: defaultNamespaceName,
 			},
 			Data: map[string]string{},
 		}
-		Expect(testEnv.Create(ctx, testConfigmap)).To(Succeed())
+		Expect(testEnv.Create(ctx, newConfigmap)).To(Succeed())
+		defer func() {
+			Expect(testEnv.Delete(ctx, newConfigmap)).To(Succeed())
+		}()
```

Comment:
Does this get run even if a later assertion fails?

Reply:
Yes, it gets run.